### ISSUE TITLE
add-on secrets are echoed (logged?)

### ISSUE THREAD

**Issue by marekbiskup** [GoogleCloudPlatform/kubernetes#8811]
By code review, I stumbled upon this code in /etc/kubernetes/kube-addons.sh:

```
# $1 string with json or yaml.
# $2 count of tries to start the addon.
# $3 delay in seconds between two consecutive tries
# $3 name of this object to use when logging about it.
function create-resource-from-string() {
  local -r config_string=$1;
  local tries=$2;
  local -r delay=$3;
  local -r config_name=$1;
  while [ ${tries} -gt 0 ]; do
    echo "${config_string}" | ${KUBECTL} create -f - && \
        echo "== Successfully started ${config_name} at $(date -Is)" && \
        return 0;
    let tries=tries-1;
    echo "== Failed to start ${config_name} at $(date -Is). ${tries} tries remaining. =="
    sleep ${delay};
  done
  return 1;
}
```

It seems that it should be 
```
  local -r config_name=$4;
```
Otherwise, the secrets are echoed. (I don't know if this is a problem)

**Comment by marekbiskup** [comment]
Being fixed by #9128

**Comment by marekbiskup** [comment]
Merged in #9128
